packages/core/admin/server/services/role.js

```diff
@@ -111,7 +111,8 @@ const find = (params = {}, populate) => {
  * @returns {Promise<array>}
  */
 const findAllWithUsersCount = async (populate) => {
-  const roles = await strapi.query('admin::role').findMany({ populate });
+  const roles = await strapi.entityService.findMany('admin::role', populate);
```

Comment:
Why the change to use the entity service? Could it be possible to keep using the query engine?

Reply:
It's just that the sanitizedQuery produces a popualte object that is compatible with the entityService. It could work with the query engine though.

Would you do `const roles = await strapi.query('admin::role').findMany({ where: populate.filters });` ?

Reply:
What format does it output? I think the query engine could take a populate argument too

Reply:
The populate object has this structure

```
{
  filters: {
    $and: [
      {
        name: {
          $contains: "value",
        },
      },
    ],
  },
}

```

Reply:
Ah, I believe those are the filters of the query but that does not include a populate query. Are you sending a populate in your api request?